libff/algebra/curves/curve_utils.tcc

```diff
@@ -40,6 +40,8 @@ decltype(((GroupT *)nullptr)->X) curve_point_y_at_x(
     const base_field x_cubed = x_squared * x;
     const base_field y_squared =
         x_cubed + (GroupT::coeff_a * x) + GroupT::coeff_b;
+    // Check that y_squared is a quadratic residue (ensuring that sqrt()
```

Comment:
Maybe mark this as a TODO.  As above, I think we need to parameterize this, so the check can be disabled (or time it to check that it is indeed trivial, but IIRC it was very costly when reading large numbers of elements).  Obviously, it is required when reading from any untrusted source, so maybe we can enable it by default.  But for trusted data, I'd be keen to have a way to disable it.

Reply:
Point taken. Added a TODO tag.